iceoryx_binding_c/test/CMakeLists.txt

```diff
@@ -44,5 +44,5 @@ iox_add_executable( TARGET                  ${PROJECT_PREFIX}_moduletests
                     FILES                   ${MODULETESTS_SRC}
                     LIBS                    ${TEST_LINK_LIBS}
                     LIBS_APPLE              dl
-                    LIBS_LINUX              acl dl pthread rt
+                    LIBS_LINUX              acl dl rt
```

Comment:
This should be fine since the tests shouldn't use the posix header directly but only by the platform abstraction which already communicates the dependency as public